modules/cluster/main.tf

```diff
@@ -69,9 +69,8 @@ module "eks" {
   enable_irsa     = true
 
   worker_groups_launch_template = var.enable_worker_group && var.enable_worker_groups_launch_template ? [
-    for subnet in(var.create_vpc ? module.vpc.public_subnets : var.subnets) :
     {
-      subnets                 = [subnet]
+      subnets                 = [var.create_vpc ? module.vpc.public_subnets : var.subnets]
```

Comment:
There are advantages to both approaches, so we should not remove one. If we can best to offer both options to the end user.

Reply:
cc @patrickleet (since you added this feature!)

Reply:
there was an issue with using one that scale-up always came from the same AZ, even if there were multiple subnets

Reply:
Is this really a concern? If an AZ goes down the ASG will spin up the nodes in the other subnets.

Reply:
yes, PVCs will all be created in the same AZ in that case, and if that zone becomes unavailable then the pods that use those PVCs can't be scheduled.

There's a flag that you can use to "balance" the nodes but it's rather ungraceful.

FWIW, reasons like this PR are why the EKS module should not be a part of the JX module.

Reply:
No worries. Will close